idaes/models_extra/co2_capture_and_utilization/unit_models/membrane_1d.py

```diff
@@ -0,0 +1,321 @@
+        """
+        self.selectivity = Var(
+            self.flowsheet().time,
+            self.mscontactor.elements,
+            self.mscontactor.feed_side.component_list,
+            self.mscontactor.feed_side.component_list,
+            initialize=1,
+            units=units.dimensionless,
+        )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            self.mscontactor.elements,
+            self.mscontactor.feed_side.component_list,
+            self.mscontactor.feed_side.component_list,
+            doc="permeance calculation",
+        )
+        def permeance_calculation(self, t, e, a, b):
+            return (
+                self.permeance[t, e, a] * self.selectivity[t, e, a, b]
+                == self.permeance[t, e, b]
+            )
+
+        p_units = feed_side_units.PRESSURE
+
+        crossover_component_list = list(
+            set(self.mscontactor.feed_side.component_list)
+            & set(self.mscontactor.sweep_side.component_list)
+        )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            self.mscontactor.elements,
+            crossover_component_list,
+            doc="permeability calculation",
+        )
+        def permeability_calculation(self, t, s, m):
+            feed_side_state = self.mscontactor.feed_side[t, s]
+            if feed_side_state.get_material_flow_basis() is MaterialFlowBasis.molar:
+                mb_units = feed_side_units.FLOW_MOLE
+                rho = self.mscontactor.feed_side[t, s].dens_mol
+            elif feed_side_state.get_material_flow_basis() is MaterialFlowBasis.mass:
+                mb_units = feed_side_units.FLOW_MASS
+                rho = self.mscontactor.feed_side[t, s].dens_mass
+            else:
+                raise TypeError(
+                    "This model only supports MaterialFlowBasis equal to molar or mass"
+                )
+
+            return self.mscontactor.material_transfer_term[
+                t, s, "feed_side", "sweep_side", m
+            ] == -units.convert(
+                (
+                    rho
+                    * self.gpu_factor
+                    * self.permeance[t, s, m]
+                    * self.cell_area
+                    * (
+                        self.mscontactor.feed_side[t, s].pressure
+                        * self.mscontactor.feed_side[t, s].mole_frac_comp[m]
+                        - units.convert(
+                            self.mscontactor.sweep_side[t, s].pressure, to_units=p_units
+                        )
+                        * self.mscontactor.sweep_side[t, s].mole_frac_comp[m]
+                    )
+                ),
+                to_units=mb_units,
+            )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            self.mscontactor.elements,
+            doc="Energy balance",
+        )
+        def energy_transfer(self, t, s):
```

Comment:
For clarity, I would call this an isothermal constraint. It is not an energy balance.

Reply:
I would also suggest changing the name of the constraint (i.e. the name of the rule/function).

Reply:
changed